### DISCUSSION TITLE
Load and Save Battle-Only Scenarios

### DISCUSSION THREAD

**Discussion by pflugs30** [ihhub/fheroes2#4401] [Ideas]
As mentioned in #4399, I have been using Battle-Only to teach my kids how to play HOMM2.  I think it's a fantastic tool, not only for training and practicing, but also for testing the development itself.  I know the Battle-Only mode takes a secondary priority to the main game development, and rightfully so.  However, as the development continues, I'd love to see further progress on this feature.

One significant improvement to the Battle-Only mode, I believe, would be the ability to save and load Battle-Only scenarios.  A Battle-Only scenario can take quite a while to setup and get exactly as one wants, particularly if one configures all possible options.  Saving and loading a scenario would allow a User to start from a template and modify as they like, saving quite a bit of time when trying several versions of a scenario.  Here again, I watched my kids create the same scenario over and over while tweaking just one or two things.

**Example Battle-Only Scenario:**
![scenario](https://user-images.githubusercontent.com/4409272/137005608-08b5add7-f3e1-4bca-b370-3f2f011f51bc.JPG)

I should mention several alternatives to this feature:
1. It is possible to save map scenarios at a difficult battle and practice by attempting and loading repeatedly.  However, that method requires having the scenario already available.  Maps can take a long time to develop.  Plus, a User may want to attempt a scenario he or she had seen in a YouTube video speedrun (true story at my house).
2. A User could also create a map from scratch using the Map Editor and build exactly the scenario they want to play.  The downside to this method is that a map will require more time and effort to build as compared to the Battle-Only configuration window.  In a word, it's overkill.

I'm interested to hear others' thoughts on this idea or where the project leaders see the Battle-Only feature going in the future.

**Comment by LeHerosInconnu** [comment]
@pflugs30.

In my list of improvements for the "Battle Only" module, as mentioned here https://github.com/ihhub/fheroes2/discussions/4399, I already have some ideas:
- To be able to load and save battle configurations in the "Battle Only" module.
- And being able to save and load a battle for replay.
- I also thought about a point system related to the "Battle Only" module.
But it's still a draft: :)

For the fight, I thought of a system where the first player to play, the attacker, is randomly designated.
Then the players first choose the type of hero (Knight, Barbarian, etc.) among those proposed and a number of points to spend is assigned to them.
Then, alternately, each player chooses one of the available proposals among, primary skills, secondary skills, spells, troops, artifacts until their points are exhausted.
As the choices are made, each player knows what the other player selects.
Then the combat begins.

Example:
Player B is designated to start the combat, he is the attacker.
He can choose between Knight, Necromancer, Sorceress or Warlock.
Player B chooses Warlock.
Player A can choose between Knight, Necromancer or Sorceress.
Player A chooses Sorceress.
Each player then receives 20,000 points to spend.
A list of at least 40,000 points is then proposed.

The defender can choose to defend in a castle:
Siege - 3000 points.

The defender may choose to increase the defense of the castle:
Defense +1 - 1000 points.
Defense +1 - 1000 points.
Defense +1 - 1000 points.
... etc.

Available for both players:
Attack +1 - 1000 points.
Attack +1 - 1000 points.
Defense +1 - 1000 points.
Defense +1 - 1000 points.
... etc.
Archery +1 - 500 points.
Archery +1 - 500 points.
Balistics +1 - 500 points.
Balistics +1 - 500 points.
... etc.
Lightning Bolt - 1000 points.
Shield - 500 points.
Teleport - 1500 points.
Haste - 500 points.
Cure - 500 points.
Anti-Magic - 1500 points.
... etc.
ranger - 20 points per unit, max 20 units.
elf - 20 points per unit, max 20 units.
wolf - 20 points per unit, max 20 units.
champion - 40 points per unit, max 10 units.
20 skeletons - 200 points.
1 genie - 400 points.
... etc.
Giant Flail of Dominion - 1000 points.
Medal of Honor - 500 points.
Enchanted Hourglass - 1000 points.
Witch's Broach of Magic - 3000 points.
... etc.

All values are only for example.

An "editor" also allows to create heroes with primary skills, secondary skills, spells, troops, artifacts already determined according to a number of points so that they are ready to face each other. :)

**Reply by pflugs30** [reply]
I think that a points system is a fantastic idea.  I can envision a lot of tournament-style scenarios that would support.  Great to have that idea documented for future development!
